content/docs/testing-recipes.md

```diff
@@ -278,126 +220,106 @@ it("changes value when clicked", () => {
 UIs could use timer based functions like `setTimeout` to run time sensitive code. In this example, a multiple choice panel waits for a selection and advances, timing out if a selection isn't made in 5 seconds.
 
 ```jsx
-function Card(props) {
-  useEffect(
-    props => {
-      const timeoutID = setTimeout(() => {
-        props.onSelect(null);
-      }, 5000);
-      return () => {
-        clearTimeout(timeoutID);
-      };
-    },
-    [props.onSelect]
-  );
-  return [1, 2, 3, 4].map(choice => {
+// card.js
+import React, { useEffect } from "react";
+
+export default function Card({ onSelect }) {
+  useEffect(() => {
+    const timeoutID = setTimeout(() => {
+      onSelect(null);
+    }, 5000);
+    return () => {
+      clearTimeout(timeoutID);
+    };
+  }, [onSelect]);
+  return [1, 2, 3, 4].map(choice => (
     <button
       key={choice}
-      data-test-id={choice}
-      onClick={() => props.onSelect(choice)}
+      data-testid={`card-button-${choice}`}
+      onClick={() => onSelect(choice)}
     >
       {choice}
-    </button>;
-  });
+    </button>
+  ));
 }
 ```
 
 We can write tests for this component by leveraging jest's timer mocks, and testing the different states it can be in.
 
+> Note: When using Jest timers, you can't rely on React Testing Library calling [`act()`](/docs/act.html) for you, so you have to call it manually. As a convenience, `@testing-library/react` exports `act`.
+
 ```jsx
 // card.test.js
-jest.useFakeTimers();
-
-let container = null
-beforeEach(() => {
-  // setup a DOM element as a render target
-  container = document.createElement('div')
-  document.body.appendChild(container)
-})
-
-afterEach(() => {
-  // cleanup on exiting
-  container.remove();
-  container = null  
-})
+import React from "react";
+import { render, fireEvent, act } from "@testing-library/react";
+import Card from "./card";
 
-function render(ui){
-  act(() => {
-    ReactDOM.render(ui, container);
-  });
-}
+jest.useFakeTimers();
 
-it("should select null after timing out", () => {
-  const yields = [];
```

Comment:
Typically it's better to use mock functions because you can assert on the calls specifically. I'd rather encourage using mock functions.